back/app/services/activities_service.rb

```diff
@@ -45,4 +47,20 @@ def create_invite_not_accepted_since_3_days_activities(now, last_time)
       LogActivityJob.perform_later(invite, 'not_accepted_since_3_days', nil, now.to_i)
     end
   end
+
+  def create_phase_ending_soon_activities(now)
+    Phase.published.where(end_at: now..now + 2.days).each do |phase|
+      if Activity.find_by(item: phase, action: 'ending_soon').nil?
+        LogActivityJob.perform_later(phase, 'ending_soon', nil, now.to_i)
+      end
+    end
+  end
+
+  def create_basket_not_submitted_activities(now)
+    Basket.not_submitted.each do |basket|
+      if Activity.find_by(item: basket, action: 'not_submitted').nil? && basket.baskets_ideas.order(:updated_at).last.updated_at <= now - 1.day
```

Comment:
I think we need to also filter out baskets from a past phase. It might happen that the basket was created right before the phase ended and that the notification is triggered after the phase ended.

Reply:
Hmm yes